tests/InstantTest.php

```diff
@@ -373,6 +374,34 @@ public function testIsBeforeOrEqualTo(int $s1, int $n1, int $s2, int $n2, int $c
         $this->assertSame($cmp <= 0, Instant::of($s1, $n1)->isBeforeOrEqualTo(Instant::of($s2, $n2)));
     }
 
+    /**
+     * @dataProvider providerInclusiveBetweenCompareTo
+     *
+     * @param int $s1  The epoch second of the 1st instant.
+     * @param int $n1  The nanosecond adjustment of the 1st instant.
+     * @param int $s2  The epoch second of the 2nd instant.
+     * @param int $n2  The nanosecond adjustment of the 2nd instant.
+     * @param int $cmp The comparison value.
+     */
+    public function testIsBetweenInclusive(int $s1, int $n1, int $s2, int $n2, int $cmp)
+    {
+        $this->assertSame($cmp <= 0, Instant::of($s1, $n1)->isBetweenInclusive(Instant::of($s2, $n2), Instant::of($s2, $n2)));
+    }
+
+    /**
+     * @dataProvider providerExclusiveBetweenCompareTo
+     *
+     * @param int $s1  The epoch second of the 1st instant.
+     * @param int $n1  The nanosecond adjustment of the 1st instant.
+     * @param int $s2  The epoch second of the 2nd instant.
+     * @param int $n2  The nanosecond adjustment of the 2nd instant.
+     * @param int $cmp The comparison value.
+     */
+    public function testIsBetweenExclusive(int $s1, int $n1, int $s2, int $n2, int $cmp)
+    {
+        $this->assertSame($cmp <= 0, Instant::of($s1, $n1)->isBetweenExclusive(Instant::of($s2, $n2), Instant::of($s2, $n2)));
```

Comment:
You're not testing much in these two methods, you're only testing that the value is within an interval of 0 seconds.

What I would do here is hardcode the interval start and end, and test with a range of possible values in between:

```php
public function testIsBetweenExclusive(int $seconds, int $nanos, bool $isBetween)
{
    $this->assertSame($isBetween, Instant::of($seconds, $nanos)->isBetweenExclusive(
        Instant::of(-1, -1),
        Instant::of(1, 1)
    );
}
```

And for the provider:

```php
[
    [-1, -2, false],
    [-1, -1, false],
    [-1,  0, true],
    [-1, 1, true],
    [0, -1, true],
    [0, 0, true],
    [0, 1, true],
    [1, -1, true],
    [1, 0, true],
    [1, 1, false],
    [1, 2, false],
]
```

Same for isBetweenInclusive(), with the data provider altered accordingly.